govcd/vm.go

```diff
@@ -1790,3 +1790,46 @@ func (client *Client) GetVMByHref(vmHref string) (*VM, error) {
 
 	return vmFunctions.GetVMByHref(client, vmHref)
 }
+
+// UpdateStorageProfile updates VM storage profile and returns refreshed VM or error.
+func (vm *VM) UpdateStorageProfile(storageProfileRef *types.Reference) (*VM, error) {
+	task, err := vm.UpdateStorageProfileAsync(storageProfileRef)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
+
+	err = vm.Refresh()
+	if err != nil {
+		return nil, err
+	}
+
+	return vm, nil
+}
+
+// UpdateStorageProfileAsync updates VM storage profile and returns Task and error.
+func (vm *VM) UpdateStorageProfileAsync(storageProfileRef *types.Reference) (Task, error) {
```

Comment:
Changed here to use just Href as string. Idea to use wider objects to avoid clashes if API requires more info and interface of function then has to change. I used new reference to make API less complain on data. Just send minimum it needs.

Reply:
We should, of course, update the signature of `UpdateStorageProfile` to use the same input

Reply:
done also